onnx/reference/ops/op_optional_has_element.py

```diff
@@ -9,11 +9,7 @@
 class OptionalHasElement(OpRun):
     def _run(self, x=None):  # type: ignore
         if x is None:
-            return ([],)
-        if isinstance(x, list):
-            if len(x) > 0:
-                return (np.array([e is not None for e in x]),)
-        elif isinstance(x, np.ndarray):
-            if len(x.shape) > 0 and x.shape[0] > 0:
-                return (np.array([e is not None for e in x]),)
-        return ([],)
+            return (np.array(False),)
+        if isinstance(x, (np.ndarray, list)):
+            return (np.array(True),)
+        return (np.array(False),)
```

Comment:
When is line 15 expected to be used? I would expect `return (np.array(x is None),)` to be sufficient ... unless you want to check whether an invalid input is supplied and raise an exception for that case, but that's a different point anyway.

Reply:
Specifically, I think raising error is better in line 15.

Reply:
done